src/Servers/Kestrel/Transport.Sockets/src/Internal/TransferResult.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal
+{
+    internal readonly struct TransferResult
+    {
+        public readonly SocketException? SocketError;
+
+        public readonly int BytesTransferred;
+
+        public readonly bool HasError;
```

Comment:
You can imply this by checking if SocketError is null

Reply:
Less state is better